common/includes/wporg-sso/wp-plugin.php

```diff
@@ -98,12 +106,25 @@ public function __construct() {
 					// Updated TOS interceptor.
 					add_filter( 'send_auth_cookies', [ $this, 'maybe_block_auth_cookies' ], 100, 5 );
 
+					// See https://core.trac.wordpress.org/ticket/61874
+					add_action( 'set_auth_cookie', [ $this, 'record_last_auth_cookie' ], 10, 6 );
+
 					// Maybe nag about 2FA
-					add_filter( 'login_redirect', [ $this, 'maybe_redirect_to_enable_2fa' ], 1000, 3 );
+					add_filter( 'login_redirect', [ $this, 'maybe_redirect_to_recovery_codes' ], 500, 3 );
+					add_filter( 'login_redirect', [ $this, 'maybe_redirect_to_enable_2fa' ], 1100, 3 );
 				}
 			}
 		}
 
+		/**
+		 * Records the last set cookies, because WordPress.
```

Comment:
🤣  I understand this. 

The documentation might be pretty confusing for someone skimming in the future.

Reply:
Totally agree :) 

```suggestion
		 * Records the last set cookies, because WordPress.
		 *
		 * During the WordPress login process, the authentication cookies are not yet available,
		 * but we need to know the user token (contained in those cookies) to retrieve their session.
		 * To work around this, we store the set authentication cookies here for later usage.
```